### DISCUSSION TITLE
Handling atomWithReset typing on v2 migration

### DISCUSSION THREAD

**Discussion by maxgertzen** [pmndrs/jotai#1883] [Q&A]
After migrating to jotai v2, I'm getting a new TS error which comes from `Args extends unknown[]`

![image](https://user-images.githubusercontent.com/34470211/232283346-090cfc12-35eb-4026-ab0d-48a9b913d397.png)

![image](https://user-images.githubusercontent.com/34470211/232283294-dfa33651-f5f8-4d86-846c-3a218d8e54b3.png)

This was implemented based on documentation - I've typed the value in the setter atom. It checks if it's of type RESET and resets the atom if needed.
But with the updated typing, it doesn't allow it. 

Should I type this differently? Or was it not covered in the latest update?

I use this resettable atom to invoke a new API call (in the getter). If I won't use it, it'll use the cached value.

Please advise. I couldn't find info in the discussion/issues.

Thank you!

**Comment by maxgertzen** [comment]
Also it messes up the useResetAtom -
![image](https://user-images.githubusercontent.com/34470211/232299120-4f8aade5-1826-4599-9018-8f99bfeebfdc.png)

**Reply by maxgertzen** [reply]
Typing this way, solved the TS issue - although not sure it's very intuitive or the solution that should be done.

![image](https://user-images.githubusercontent.com/34470211/232307006-4483e573-435f-45f8-a7fa-dd83176e7516.png)
![image](https://user-images.githubusercontent.com/34470211/232307013-84fa7931-ea9a-4074-ab02-5d4912f460c2.png)

**Comment by dai-shi** [comment]
It should be like `atom<OtpVerificationPageData | null, [typeof RESET], void>`.
But, I'd recommend to remove generics types and only type the argument.
```ts
  async (get, set, isReset: typeof RESET) => {
```

Can anyone help updating docs?

**Reply by maxgertzen** [reply]
Thank you @dai-shi !
Your recommendation works but requires updating the `useResetAtom` type/signature as well
![image](https://user-images.githubusercontent.com/34470211/232312925-0f621dc0-3246-477b-83bc-3f53c295c94d.png)

**Reply by dai-shi** [reply]
> Type 'unknown' is not assignable to type 'Promise<void>'.

Can someone familiar with TS look into it? Maybe, we need `any` instead of `unknown`.
